components/gantry/oneaxis/oneaxis.go

```diff
@@ -160,13 +164,65 @@ func newOneAxis(ctx context.Context, deps resource.Dependencies, conf resource.C
 		return nil, errors.Errorf("invalid gantry type: need 1, 2 or 0 pins per axis, have %v pins", np)
 	}
 
-	if err := oAx.home(ctx, np); err != nil {
+	if err := oAx.home(context.Background(), np); err != nil {
 		return nil, err
 	}
 
+	oAx.checkHit(context.Background())
 	return oAx, nil
 }
 
+func (g *oneAxis) checkHit(ctx context.Context) {
+	ctx, g.cancelFunc = context.WithCancel(ctx)
+	g.activeBackgroundWorkers.Add(1)
+	utils.PanicCapturingGo(func() {
+		defer utils.UncheckedErrorFunc(func() error {
+			return nil
+		})
+		defer g.activeBackgroundWorkers.Done()
+		for {
+			if ctx.Err() != nil {
+				return
+			}
+			select {
+			case <-ctx.Done():
+				return
+			default:
+			}
+			func() {
+				g.mu.Lock()
+				defer g.mu.Unlock()
+				if hit0, _ := g.limitHit(ctx, true); hit0 {
+					g.moveAway(ctx, true)
+				}
+				if hit1, _ := g.limitHit(ctx, false); hit1 {
+					g.moveAway(ctx, false)
+				}
+			}()
+		}
+	})
+}
+
+func (g *oneAxis) moveAway(ctx context.Context, zero bool) {
+	g.motor.Stop(ctx, nil)
+	time.Sleep(100 * time.Millisecond)
+	if zero {
+		_ = g.motor.GoFor(ctx, g.rpm, 0, nil)
```

Comment:
GoFor with zero revolutions passed in goes forever. Do we want that? Or are you trying to go fore until not hit somehow? Which would be really cool.

Also, we should always check errors otherwise we get stuck in the loop forever.

Reply:
yes that, go "forever" until the limit switch is no longer hit, so forever is practically nothing

Reply:
really elegant thank you, but also - please add a comment about the physical behaviour of this function